project_template/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX
+    VERSION 0.1.0)
+
+option(BUILD_DOCS "Build documentation" ON)
+option(BUILD_TESTING "Build tests" ON)
+set(AZ_INSTALL_CMAKEPACKAGEDIR "share/cmake" CACHE STRING "directory to install cmake packages into")
+
+# if needed: add ./cmake to the module path
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 17)
+
+# default to hidden symbols when building shared libraries
+# on unix systems
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
+
+set_property(DIRECTORY PROPERTY COMPILE_OPTIONS
+    $<$<COMPILE_LANG_AND_ID:C,MSVC>:/W4 /wd4201>
+    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Xclang -Wall -Wextra -Wno-microsoft-anon-tag -Wdocumentation>)

```

Comment:
what is going to be our strategy for cross compilation?

Reply:
cmake toolchain files.